File: fortunecards.client/src/app/models/deck.ts
import { Card } from './card';

export interface Deck {
  id: number;
  name: string;
  description: string | null;
  createdAt: string;
  emoji: string;
  colorIndex: number;
  cardBackImageUrl: string | null;
  aspectWidth: number;
  aspectHeight: number;
  cardCount?: number;
  cards?: Card[];
  isPublic: boolean;
  isOwner: boolean;
  isFavorite: boolean;
}

export interface CreateDeckPayload {
  name: string;
  description: string | null;
  emoji: string;
  colorIndex: number;
  aspectWidth: number;
  aspectHeight: number;
  isPublic: boolean;
  cardBackImage?: File;
}

File: FortuneCards.Server/Models/Deck.cs
namespace FortuneCards.Server.Models
{
    public class Deck
    {
        public int Id { get; set; }
        public required string Name { get; set; }
        public string? Description { get; set; }
        public string Emoji { get; set; } = "🎴";
        public int ColorIndex { get; set; } = 0;
        public string? CardBackImageUrl { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public ICollection<Card> Cards { get; set; } = new List<Card>();
    }
}
